assembly/pallets/balances/balances.ts

```diff
@@ -38,40 +41,62 @@ export class Balances {
         if(BalancesConfig.existentialDeposit().unwrap() > freeBalance.unwrap()) {
             return ResponseCodes.VALIDITY_ERROR;
         }
-        BalancesStorageEntries.Account().set(freeBalance, accountId);
+        const accData = BalancesStorageEntries.Account().get(accountId);
+        accData.setFree(freeBalance);
+        accData.setReserved(reservedBalance);
+        BalancesStorageEntries.Account().set(accData, accountId);
         Balances._depositEvent(EventTypes.BalanceSet, accountId.toU8a().concat(freeBalance.toU8a()));
         return ResponseCodes.SUCCESS;
     }
 
+    /**

```

Comment:
Can we add info here that it is only for correct metadata generation and for compliance with enforced limits from PolkadotJS